File: Java/Linked List/LinkedList.java
public class LinkedList {
	Node head;

	static class Node {
		int data;
		Node next;

		Node(int x) {
		
		data = x;
		next = null;
		}

	}

	void push(int value_toPush) {

		Node newNode = new Node(value_toPush);

		if (head == null) {
			head = newNode;
			return;
		}

		newNode.next = head;
		head = newNode;
	}

	void append(int value_toAppend) {

		Node newNode = new Node(value_toAppend);

		if (head == null) {
			head = newNode;
			return;
		}

		Node last = head;

		while (last.next != null)
			last = last.next;

		last.next = newNode;
	}

	void remove() {

		Node currentNode, previousNode;

		currentNode = previousNode = head;

		while (currentNode.next != null) {
			previousNode = currentNode;
			currentNode = currentNode.next;
		}

		System.out.println("Item Deleted: " + currentNode.data);
		
		if (currentNode == head) {
			head = currentNode.next;
			currentNode = null;
			return;
		}

		currentNode = null;
		previousNode.next = null;
	}

	void remove(int value_toRemove) {

		Node currentNode, previousNode;

		currentNode = previousNode = head;

		while (currentNode.data != value_toRemove) {
			previousNode = currentNode;
			currentNode = currentNode.next;
		}

		if (currentNode == null) {
			System.out.println("Item Not Found!!!");
			return;
		}
		
		System.out.println("Item Delete: " + currentNode.data);

		if (currentNode == head) {
			head = currentNode.next;
			currentNode = null;
			return;
		}

		previousNode.next = currentNode.next;
		currentNode = null;
	}

	void printList() {

		Node temp = head;

		while (temp != null) {
		
			System.out.print(temp.data + " ");
			temp = temp.next;
		}
		System.out.println();
	}
	
	public static void main(String []args) {

		LinkedList lList = new LinkedList();

		lList.push(3);
		lList.push(2);
		lList.push(1);

		lList.printList();

		lList.append(4);
		lList.append(5);
		lList.append(6);

		lList.printList();

		lList.remove();

		lList.printList();

		lList.remove();

		lList.printList();

		lList.remove(3);
		
		lList.printList();

		lList.remove(1);

		lList.printList();
		
	}
}

File: Python/Linked List/LinkedList.py
class Node:
	def __init__(self, data):
		self.data = data
		self.next = None

		
class LinkedList:
	def __init__(self):
		self.head = None
		
	def push(self, value_toInsert):
		new_Node = Node(value_toInsert)
		
		if self.head == None:
			self.head = new_Node
			
		else:
			new_Node.next = self.head
			
			self.head = new_Node
			
	def append(self, value_toAppend):
		new_Node = Node(value_toAppend)
		
		if self.head is None:
			self.head = new_Node
			return
			
		temp = self.head
		
		while (temp.next):
			temp = temp.next
			
		temp.next = new_Node

	def remove(self):
		currentNode = previousNode = self.head
		
		while currentNode.next:
			previousNode = currentNode
			currentNode = currentNode.next
		
		if currentNode is None:
			print ("Empty List");
			return;
			
		print ("Item Deleted: {}".format(currentNode.data))	
			
		if currentNode == self.head:
			self.head = currentNode.next
			del(currentNode)
			return;
			
		previousNode.next = None;
		del(currentNode)
		
	def removeValue(self, value_toRemove):
		currentNode = previousNode = self.head
		
		while currentNode.data != value_toRemove:
			previousNode = currentNode
			currentNode = currentNode.next
		
		if currentNode is None:
			print ("Empty List");
			return;
			
		print ("Item Deleted: {}".format(currentNode.data))	
			
		if currentNode == self.head:
			self.head = currentNode.next
			del(currentNode)
			return;
			
		previousNode.next = currentNode.next;
		del(currentNode)
		
	def Display(self):
		temp = self.head

		print ("\nList: ", end="")
		
		while (temp):
			print(temp.data, " ", end="")
			temp = temp.next
			
		print()